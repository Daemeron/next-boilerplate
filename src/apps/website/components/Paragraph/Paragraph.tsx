import styled from 'styled-components';
import { ThemePropsInterface } from '../../core/theme/Theme';

export const Paragraph = styled.p<ThemePropsInterface>`
  font-size: 1rem;
  line-height: 1.4rem;
`;
