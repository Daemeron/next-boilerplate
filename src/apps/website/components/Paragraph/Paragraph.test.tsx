import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { Paragraph } from './Paragraph';

describe('Paragraph', () => {
  test('matches the snapshot', () => {
    const paragraph = render(<Paragraph>test</Paragraph>);
    expect(paragraph.container).toMatchSnapshot();
  });
  test('sets font size and line-height', () => {
    const paragraph = render(<Paragraph />);
    expect(paragraph.container.firstChild).toHaveStyleRule('font-size', '1rem');
    expect(paragraph.container.firstChild).toHaveStyleRule('line-height', '1.4rem');
  });
});
