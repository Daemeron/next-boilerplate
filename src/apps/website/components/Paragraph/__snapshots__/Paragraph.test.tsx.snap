// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph matches the snapshot 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.4rem;
}

<div>
  <p
    class="c0"
  >
    test
  </p>
</div>
`;
